File: codility/lessons/2 - arrays/odd occurrences in array/odd_occurences_in_array_procedural.cs
// Source: https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/

using System;

class Solution
{
    public int solution(int[] A)
    {
        Array.Sort(A);
        int unpairedElement = A[0];
        bool isElementPaired = false;

        for (int i = 1; i < A.Length; i++)
            if (A[i] == unpairedElement)
                isElementPaired = !isElementPaired;
            else if (isElementPaired)
            {
                isElementPaired = false;
                unpairedElement = A[i];
            }
            else
                break;

        return unpairedElement;
    }
}

File: codility/lessons/2 - arrays/odd occurrences in array/odd_occurences_in_array_procedural.cpp
// Source: https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/

#include  <algorithm>

int solution(vector<int> &A)
{
    ranges::sort(A);
    int unpaired_element {A.front()};
    bool is_element_paired {false};

    for (size_t i {1}; i < A.size(); ++i)
        if (A.at(i) == unpaired_element)
            is_element_paired = {!is_element_paired};
        else if (is_element_paired) {
            is_element_paired = {false};
            unpaired_element = {A.at(i)};
        }
        else
            break;

    return unpaired_element;
}
